### DISCUSSION TITLE
how to change namespace

### DISCUSSION THREAD

**Discussion by devcui** [fabric8io/kubernetes-client#3860] [Q&A] [answered]
```java
 InputStream serviceMonitorIn = new ByteArrayInputStream(serviceMonitor.get(0).getBytes(StandardCharsets.UTF_8));
            CustomResourceDefinition crd = this.kubernetesClient
                    .inNamespace("monitoring")
                    .apiextensions()
                    .v1()
                    .customResourceDefinitions()
                    .load(serviceMonitorIn)
                    .get();
```

is my code, and the `DefaultKubernetesClient` namespace was `cloud`, when i exec the code i will have an error

`io.fabric8.kubernetes.client.KubernetesClientException: Namespace mismatch. Item namespace:monitoring. Operation namespace:cloud.`

how to change the namespace of the DefaultKubernetesClient?

**Comment by rohanKanojia** [comment, marked as answer]
I think client seems to be complaining for difference in operation namespace (`inNamespace(monitoring)`) and namespace present in `metadata.namespace` content of `serviceMonitorIn`
